In [ ]:
from google.colab import files
uploaded = files.upload()  # ты сам выбираешь файл
filename = list(uploaded.keys())[0]
df = pd.read_csv(filename)  # или pd.read_excel

Saving Chocolate_Sales.csv to Chocolate_Sales (1).csv


In [ ]:
import pandas as pd

In [ ]:
df = pd.read_csv('Chocolate_Sales.csv')

In [ ]:
# Посмотрим первые 5 строк
print("Первые 5 строк:")
print(df.head())

# Посмотрим информацию о данных
print("\nИнформация о данных:")
print(df.info())

# Посмотрим описательную статистику
print("\nОписательная статистика:")
print(df.describe())

Первые 5 строк:
     Order_ID            Product    Country Channel    Salesperson  \
0  ORD-069833   Truffle Gift Box  Australia  Retail    Arjun Mehta   
1  ORD-090726       85% Dark Bar  Australia  Retail    Arjun Mehta   
2  ORD-042159       70% Dark Bar      Japan  Retail  Hannah Müller   
3  ORD-197166  Hazelnut Milk Bar    Germany  Retail    Arjun Mehta   
4  ORD-112162  Almond Crunch Bar  Australia  Retail      Yuki Sato   

   Order_Date  Discount_Pct  Price_per_Box  Marketing_Spend  Boxes_Shipped  \
0  2022-12-11           3.5          13.72           202.03             71   
1  2023-03-14           9.4           3.30            55.18             84   
2  2023-12-21           4.9          18.21            60.65             35   
3  2023-12-18          15.0           2.66            52.00             92   
4  2023-08-18           4.4           2.75           187.44            214   

   Amount  
0  912.31  
1  245.91  
2   583.7  
3  211.27  
4  549.69  

Информация о данных:


In [ ]:
# Сколько пропусков в каждом столбце

print('\nПропуски в данных')
print(df.isnull().sum())


Пропуски в данных
Order_ID             0
Product              0
Country              0
Channel              0
Salesperson          0
Order_Date         437
Discount_Pct       489
Price_per_Box      457
Marketing_Spend    461
Boxes_Shipped        0
Amount               0
dtype: int64


In [ ]:
# Очищаем колонку Amount
# 1. Убираем символы $ и пробелы
df['Amount'] = df['Amount'].astype(str).str.replace('$', '').str.strip()

# 2. Заменяем пустые строки на NaN
df['Amount'] = df['Amount'].replace('', pd.NA)

# 3. Преобразуем в числа (если не получается — ставим NaN)
df['Amount'] = pd.to_numeric(df['Amount'], errors='coerce')

# 4. Удаляем строки, где Amount стал NaN (или можно заполнить)
df = df.dropna(subset=['Amount'])

# Проверяем, что получилось
print(f"Осталось строк после очистки: {len(df)}")
print(df[['Amount']].head(10))

Осталось строк после очистки: 200000
   Amount
0  912.31
1  245.91
2  583.70
3  211.27
4  549.69
5  527.01
6  470.63
7  629.51
8  181.16
9  588.18


In [ ]:
# Ключевые метрики

# 2. Проверяем колонки
print("Доступные колонки:", list(df.columns))

# 3. Выбираем колонку для анализа
amount_col = input("Введите название колонки с суммой: ")

# 4. Очищаем данные
df[amount_col] = pd.to_numeric(df[amount_col], errors='coerce')
df = df.dropna(subset=[amount_col])

# 5. Считаем метрики
total = df[amount_col].sum()
average = df[amount_col].mean()
orders = len(df)

# 6. Выводим с округлением
print("\n" + "="*40)
print(f"Общая выручка: {total/1_000_000:.2f} млн руб.")
print(f"Средний чек: {average:,.2f} руб.")
print(f"Количество заказов: {orders:,}")

Доступные колонки: ['Order_ID', 'Product', 'Country', 'Channel', 'Salesperson', 'Order_Date', 'Discount_Pct', 'Price_per_Box', 'Marketing_Spend', 'Boxes_Shipped', 'Amount']
Введите название колонки с суммой: Amount

Общая выручка: 102.57 млн руб.
Средний чек: 512.86 руб.
Количество заказов: 200,000
Общая выручка: 102.57 млн руб.


## Выводы после первичного анализа:
- Всего строк: 200000
- Количество столбцов: 11
- Есть пропуски в колонках: Order_Date, Discount_Pct, Price_per_Box,    Marketing_Spend
- Основные метрики: общая выручка ~ 102,572.4 тыс. руб, средний чек ~ 512.86 руб.

In [ ]:
df.to_csv('chocolate_sales_clean.csv', index=False)
files.download('chocolate_sales_clean.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>